### DISCUSSION TITLE
`grantPermission` doesn't work for Photo permission

### DISCUSSION THREAD

**Discussion by yutanui** [leancodepl/patrol#563] [Q&A] [answered]
Hi Patrol team,

Thank you very much for Patrol package.

I have checked and tried to utilize the package instead of integration_test for permission grant.

Anyway, With your example, I tried to uitlize it but still cannot perform dialog permission for photos such as (All Photos) with iOS simulator (I have tested with iOS 15.0)

Sample of code as below.  (I have set up all required configuration for permission_handler library already)

```
Future<void> requestAndGrantPhotosPermission(PatrolTester $) async {
    expect($(#photo).$(#statusText).text, 'Not granted');

    await $('Request photo permission').tap();

    final request = await $.native.isPermissionDialogVisible(
      timeout: Duration(seconds: 2),
    );
    if (request) {
      await $.native.grantPermissionWhenInUse();
    }

    await $.pump();
    expect($(#photo).$(#statusText).text, 'Granted');
  }
```

Could you please suggest what resource should I looking for or provide an example of photos permission with Patrol package ?

Thank you very much

**Comment by bartekpacia** [comment, marked as answer]
Hi @yutanui,

What language do you have set in your iOS Simulator? Currently granting permissions works only in English. If you're using a different language, you have to use `$.native.tap` manually with the text.

**Reply by yutanui** [reply]
Hi @bartekpacia ,

I use English for the simulator.

Do you mean something like this ?
`await $.native.tap(Selector(text: 'Allow Access to All Photos'));`

I have tried but have no idea. The screenshot from the simulator as follow.

![image](https://user-images.githubusercontent.com/13654249/200004517-b12a7e70-dc5e-41dc-8816-d2e62f259887.png)


The exception in running terminal as  below.

_The following PatrolActionException was thrown running a test:
Patrol action failed: tap() failed with code NOT_FOUND (view with text "Allow Access to All Photos"
in app "pl.leancode.patrol.Example" doesn't exist)_


Thank you very much.

**Reply by bartekpacia** [reply]
Yes, I mean exactly this:

```dart
await $.native.tap(Selector(text: 'Allow Access to All Photos'));
```

There's just one thing you're missing: you need to specify the correct `bundleId` of the app you're tapping on. The permission request dialog you're seeing doesn't belong to `pl.leancode.patrol.Example`, but to `com.apple.springboad` (the iOS launcher). That's why an error is thrown.

Please do:

```dart
await $.native.tap(
  Selector(text: 'Allow Access to All Photos'),
  appId: 'com.apple.springboard',
);
```

Let me know if this helps! :)

**Reply by yutanui** [reply]
Yes, It's work !!! Thank you so much

**Reply by bartekpacia** [reply]
Happy to help :)
